src/graph/unit_graph.cc

```diff
@@ -1152,4 +1152,38 @@ SparseFormat UnitGraph::SelectFormat(SparseFormat preferred_format) const {
     return SparseFormat::COO;
 }
 
+bool UnitGraph::Load(dmlc::Stream* fs) {
+  int64_t num_vtypes, num_src, num_dst;
+  CHECK(fs->Read(&num_vtypes)) << "Invalid num_vtypes";
+  CHECK(fs->Read(&num_src)) << "Invalid num_src";
+  CHECK(fs->Read(&num_dst)) << "Invalid num_dst";
+  aten::CSRMatrix csr_matrix;
+  CHECK(fs->Read(&csr_matrix)) << "Invalid csr_matrix";
+  SparseFormat restrict_format;
+  CHECK(fs->Read(&restrict_format)) << "Invalid restrict_format";
+  auto mg = CreateUnitGraphMetaGraph(num_vtypes);
+  CSRPtr csr(new CSR(mg, num_src, num_dst, csr_matrix.indptr, csr_matrix.indices, csr_matrix.data));
+  in_csr_ = csr;
+  meta_graph_ = mg;
+  restrict_format_ = restrict_format;
+  // HeteroGraphPtr hptr = CreateFromCSR(num_vtypes, num_src, num_dst, csr_matrix.indptr,
```

Comment:
The code aboves is copied from `CreateFromCSR`. I want to reuse the code in `CreateFromCSR`, but due to my limited C++ skill I don't know how to properly move `hptr` into `this`. Do I need something like a move constructor? Could you give any suggestions? @BarclayII @jermainewang

Reply:
I would write `*this = UnitGraph(mg, csr, nullptr, nullptr, restrict_format)`.